### ISSUE TITLE
[Bug]: 6.6.1 and 6.6.2 the backslash character is doubled in the example

### ISSUE THREAD

**Issue by pairbit** [domaindrivendev/Swashbuckle.AspNetCore#2967]
### Describe the bug

Commentary on the models.
The comment contains line breaks (\r\n)

```csharp
/// <example><![CDATA[<Doc>Multi\r\nline text</Doc>]]></example>
public string Data { get; set; }
```

The swagger displays already doubled backslash symbols

```xml
[
  {
    "data": "<Doc>Multi\\r\\nline text</Doc>"
  }
]
```

### Expected behavior

In a swagger, backslash characters should not be doubled

```xml
[
  {
    "data": "<Doc>Multi\r\nline text</Doc>"
  }
]
```

### Swashbuckle.AspNetCore version

6.6.1 and 6.6.2

### .NET Version

net8.0

**Comment by RaphaelFakhri** [comment]
I do not think this is a bug, and I think 6.6.1 was the fix rather than the break.

I ran the real `XmlCommentsSchemaFilter` over a property carrying exactly the doc comment in the report and serialized the result with the library's own writers. On current master:

```json
{
  "type": "string",
  "example": "<Doc>Multi\\r\\nline text</Doc>"
}
```

and with the YAML writer:

```yaml
example: <Doc>Multi\r\nline text</Doc>
```

The in-memory value is `<Doc>Multi\r\nline text</Doc>` with two literal backslashes, and parsing the emitted JSON back gives the same thing.

That is correct. The doc comment declares a string containing the two characters backslash and `r`, not a carriage return. JSON requires a backslash to be written as `\\`, so `"...\\r\\n..."` is the only valid encoding of that value, which is why the YAML writer, which needs no such escaping, prints single backslashes. The expected output in the report, `"<Doc>Multi\r\nline text</Doc>"`, decodes to a string containing a real CR and LF, which is a different value from the one the source declares.

The behaviour change in 6.6.1 was #2727. Before it, `TrySetExample` built the JSON by concatenation, `$"\"{example}\""`, and handed that to a `JsonElement` deserializer, so the source text's `\r` and `\n` were interpreted as JSON escapes and silently became a real carriage return and line feed. #2727 replaced that with proper serialization, which preserves the literal characters. The current equivalent is `JsonValue.Create(exampleString)` in `XmlCommentsExampleHelper.cs`.

Worth saying explicitly if this gets closed: if the intent really was a line break in the example, the way to get one is a real line break in the doc comment, not the two-character sequence `\r\n`.
